src/test/javascript/spec/component/learning-paths/components/learning-path-exercise.component.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { LearningPathExerciseComponent } from 'app/course/learning-paths/components/learning-path-exercise/learning-path-exercise.component';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TranslateService } from '@ngx-translate/core';
+
+describe('LearningPathExerciseComponent', () => {
+    let component: LearningPathExerciseComponent;
+    let fixture: ComponentFixture<LearningPathExerciseComponent>;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [LearningPathExerciseComponent],
+            providers: [{ provide: TranslateService, useValue: jest.fn() }],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(LearningPathExerciseComponent);
+        component = fixture.componentInstance;
+        fixture.componentRef.setInput('courseId', 1);
+        fixture.componentRef.setInput('exerciseId', 1);
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        expect(component).toBeTruthy();
+        expect(component.courseId()).toBe(1);
+        expect(component.exerciseId()).toBe(1);
+    });
```

Comment:
Maybe for this component, you could test whether the parameters are correctly passed to the (dynamically created) subcomponent

Reply:
In general this should not be the end solution as this is rather suboptimal. To improve this I will need to change the CourseExerciseDetailsComponent